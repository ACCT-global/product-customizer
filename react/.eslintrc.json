{
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "rules": {
    "jsx-a11y/alt-text": "off"
  }
}
