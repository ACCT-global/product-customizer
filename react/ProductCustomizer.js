import React, { Component } from 'react'
import { Spinner } from 'vtex.styleguide'
import { head } from 'ramda'

import ProductDetails from 'vtex.product-details/ProductDetails'
import ProductCustomizer from './components/ProductCustomizer'

class ProductCustomizerIndex extends Component {
  render() {
    const { productQuery: { loading, product } } = this.props

    if (loading) return (
      <div className="flex justify-center pa8 w-100">
        <Spinner />
      </div>
    )

    let hasSchema
    const { calculatedAttachments } = head(product.items)

    try {
      const schema = JSON.parse(calculatedAttachments)
      hasSchema = Object.keys(schema.properties).length > 0
    } catch (e) {
      hasSchema = false
    }

    if (!hasSchema) {
      return <ProductDetails {...this.props} />
    }

    return <ProductCustomizer {...this.props} />
  }
}

export default ProductCustomizerIndex
