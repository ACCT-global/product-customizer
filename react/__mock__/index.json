{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "http://json-schema.org/draft-07/schema#",
  "title": "Pizza Builder",
  "type": ["object", "boolean"],
  "required": ["Size","Dough","Toppings"],
  "properties": {
    "Size": {
      "description": "Escolha o tamanho",
      "type": "string", "enum":["4755","7837","7856"],
      "minItems": 1,
      "maxItems": 1,
      "uniqueItems": true
    },
    "Dough": {
      "description": "Escolha 1 tipo de massa",
      "type": "string", "enum":["4872","5698"],
      "minItems": 1,
      "maxItems": 1,
      "uniqueItems": true
    },
    "Toppings": {
      "description": "Escolha até 2 ingredientes",
      "type": "array", "items": {
        "type": "string", "enum":["Pepperoni","Meat","Bacon"]
      },
      "minItems": 1,
      "maxItems": 2,
      "uniqueItems": false
    }
  },
  "items":
  [
    {
    "Id":"4755",
      "Name": "Broto",
      "price": 18.00,
      "image": "https://via.placeholder.com/500x500",
      "pricetable": "tabelacombo"
    },
    {
    "Id":"7837",
      "Name": "Média",
      "price": 25.00,
      "image": "https://via.placeholder.com/500x500",
      "pricetable": "tabelacombo"
    },
    {
    "Id":"7856",
      "Name": "Grande",
      "price": 32.00,
      "image": "https://via.placeholder.com/500x500",
      "pricetable": "tabelacombo"
    },
    {
    "Id":"4872",
      "Name": "Fina",
      "price": 1.00,
      "image": "https://via.placeholder.com/500x500",
      "pricetable": "tabelacombo"
    },
    {
    "Id":"5698",
      "Name": "Crocante",
      "price": 2.00,
      "image": "https://via.placeholder.com/500x500",
      "pricetable": "tabelacombo"
    }
  ]
}
