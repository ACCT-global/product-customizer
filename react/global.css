@media screen and (max-width: 40em) {
  .vtex-modal__overlay {
    align-items: stretch !important;
  }

  .vtex-modal__modal {
    border-radius: 0;
  }

  .vtex-modal__close-icon {
    z-index: 10;
  }
}

@media screen and (min-width: 40em) {
  .vtex-modal__modal {
    max-height: 800px;
  }

  .vtex-product-customizer__options {
    background-color: transparent;
  }
}

.vtex-modal__modal,
.vtex-modal__overlay {
  padding: 0 !important;
}

.vtex-modal__overlay {
  background: rgba(0, 0, 0, 0.9) !important;
}

.vtex-modal__modal {
  width: 100%;
  height: 100%;
  overflow: auto;
}

.vtex-product-customizer__actions .vtex-button {
  width: 100%;
}

.vtex-product-customizer__actions .actions--change-toppings .vtex-button {
  padding: 0;
  text-align: left;
}

.vtex-product-customizer__single-choice .single-choice__image-container {
  color: #1346d8;
  max-width: 72px;
  position: relative;
}

.single-choice__description {
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 200px;
  overflow: hidden;
}

.vtex-product-customizer__single-choice.selected .single-choice__image-container .single-choice__icon-container {
  display: block;
}

.vtex-product-customizer__single-choice.selected .single-choice__image-container .vtex__icon-success {
  right: 5px;
  bottom: 10px;
  position: absolute;
}

.vtex-product-customizer__single-choice.selected .single-choice__image-container img {
  width: 100%;
}
