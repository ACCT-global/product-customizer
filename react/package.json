{
  "dependencies": {
    "classnames": "^2.2.6",
    "ramda": "^0.26.1",
    "react-intl": "^2.7.2",
    "smoothscroll-polyfill": "^0.4.3"
  }
}
