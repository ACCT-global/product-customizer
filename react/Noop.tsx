import { FC } from 'react'

const Noop: FC = () => null

export default Noop