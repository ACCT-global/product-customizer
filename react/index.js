import React, { Component } from 'react'
import { Spinner } from 'vtex.styleguide'
import { path } from 'ramda'

import { ExtensionPoint } from 'vtex.render-runtime'

import ProductCustomizer from './components/ProductCustomizer'

class ProductCustomizerIndex extends Component {

  hasAttachments = () => {
    const metadataItems = path(['product', 'itemMetadata', 'items'], this.props.productQuery)
    if (!metadataItems) { return false }
    return metadataItems.some(({ assemblyOptions }) => assemblyOptions && assemblyOptions.length > 0)
  }

  render() {
    const { productQuery: { loading } } = this.props


    if (loading) return (
      <div className="flex justify-center pa8 w-100">
        <Spinner />
      </div>
    )

    if (this.hasAttachments()) {
      return <ProductCustomizer {...this.props} />
    }

    return <ExtensionPoint id="product-details" {...this.props} />
  }
}

export default ProductCustomizerIndex
