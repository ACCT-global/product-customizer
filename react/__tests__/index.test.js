import React from 'react'
import { render, fireEvent } from '@vtex/test-tools/react'

import ProductCustomizer from '../index'
import productQueryWithVariation from '../__mocks__/productQueryWithVariation.json'
import productQueryNoVariation from '../__mocks__/productQuery.json'

describe('ProductCustomizer components ', () => {
  it('render with variations example, show sku selector', () => {
    const productQuery = { ...productQueryWithVariation.data }
    const component = render(<ProductCustomizer productQuery={productQuery} />)
    expect(component).toBeDefined()
    expect(component.asFragment()).toMatchSnapshot()
  })
  it('render with no variations and press on Small SKU', () => {
    const productQuery = { ...productQueryNoVariation.data }
    const component = render(<ProductCustomizer productQuery={productQuery} />)
    expect(component).toBeDefined()
    const { getByText } = component
    expect(component.asFragment()).toMatchSnapshot()

    const smallButton = getByText('Small')
    fireEvent.click(smallButton)
    expect(component.asFragment()).toMatchSnapshot()
  })
})
