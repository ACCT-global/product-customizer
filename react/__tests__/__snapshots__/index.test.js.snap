// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCustomizer components  render with no variations and press on Small SKU 1`] = `
<DocumentFragment>
  <div
    class="flex-ns pv6-ns justify-center-ns"
  >
    <div
      class="w-100 w-third-ns flex tc items-center-ns h-100-ns ph2-ns"
    >
      <img
        alt="Suprema"
        class="image"
        src="https://delivery.vteximg.com.br/arquivos/ids/155442/super-supreme.e34cdf7984831c25c4eda14a2d361a20.1.jpg?v=636770452538570000"
        style="object-fit: cover;"
      />
    </div>
    <div
      class="w-40-ns ph2-ns"
    >
      <div
        class="productName t-heading-4 c-on-base ph5 pt5-s"
      >
        Super Suprema
      </div>
      <div
        class="ph5 pt4 pb2 pb4-ns"
      />
      <div
        class="ph5 pb4 c-muted-2 t-small"
      >
        <span>
          Pick one size
        </span>
      </div>
      <div
        class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Small
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div
        class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Medium
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div
        class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Large
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div />
      <div
        class="w-100"
      />
      <div
        class="z-2 fixed bg-white bottom-0 w-40-ns ph1-ns w-100 bt b--light-gray"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductCustomizer components  render with no variations and press on Small SKU 2`] = `
<DocumentFragment>
  <div
    class="flex-ns pv6-ns justify-center-ns"
  >
    <div
      class="w-100 w-third-ns flex tc items-center-ns h-100-ns ph2-ns"
    >
      <img
        alt="Suprema"
        class="image"
        src="https://delivery.vteximg.com.br/arquivos/ids/155442/super-supreme.e34cdf7984831c25c4eda14a2d361a20.1.jpg?v=636770452538570000"
        style="object-fit: cover;"
      />
    </div>
    <div
      class="w-40-ns ph2-ns"
    >
      <div
        class="productName t-heading-4 c-on-base ph5 pt5-s"
      >
        Super Suprema
      </div>
      <div
        class="ph5 pt4 pb2 pb4-ns"
      >
        <div>
          ProductPrice
        </div>
      </div>
      <div
        class="ph5 pb4 c-muted-2 t-small"
      >
        <span>
          Pick one size
        </span>
      </div>
      <div
        class="selected bg-muted-5 singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Small
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div
        class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Medium
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div
        class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
        role="button"
        tabindex="0"
      >
        <div
          class="relative flex items-center justify-between pv5"
        >
          <div
            class="itemDescriptionContainer itemDescriptionContainerSingle flex"
          >
            <div />
            <div
              class="flex flex-column justify-center"
            >
              <div
                class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
              >
                Large
              </div>
              <div
                class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
              >
                <div>
                  ProductPrice
                </div>
              </div>
            </div>
          </div>
          <div
            class="singleChoiceActions mt3"
          >
            <div>
              Radio
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class=""
        >
          <div
            class="ph5 pv4 c-muted-2 t-small"
          >
            <span>
              Pick one crust
            </span>
          </div>
          <div
            class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
            role="button"
            tabindex="0"
          >
            <div
              class="relative flex items-center justify-between pv5"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerSingle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155411/cheesy-bites.png?v=636759865607400000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
                  >
                    Cheesy Bites
                  </div>
                  <div
                    class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="singleChoiceActions mt3"
              >
                <div>
                  Radio
                </div>
              </div>
            </div>
          </div>
          <div
            class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
            role="button"
            tabindex="0"
          >
            <div
              class="relative flex items-center justify-between pv5"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerSingle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155412/stuffed-crust.png?v=636759865852770000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
                  >
                    Stuffed Crust
                  </div>
                  <div
                    class="itemDescription itemDescriptionSingle flex t-small fw5 c-action-primary mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="singleChoiceActions mt3"
              >
                <div>
                  Radio
                </div>
              </div>
            </div>
          </div>
          <div
            class="false singleChoiceContainer hover-bg-muted-5 ph4 pointer bb b--muted-5 bw1"
            role="button"
            tabindex="0"
          >
            <div
              class="relative flex items-center justify-between pv5"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerSingle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155413/classic.png?v=636759866429100000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameSingle t-heading-5 c-on-base"
                  >
                    Classic Crust
                  </div>
                </div>
              </div>
              <div
                class="singleChoiceActions mt3"
              >
                <div>
                  Radio
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="dn"
        >
          <div
            class="ph5 pv4 c-muted-2 t-small"
          >
            <span>
              Basic ingredients
            </span>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155431/PIMIENTO-VERDE.png?v=636766050771230000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Green Pepper
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Green Pepper","checked":true}
                </div>
              </div>
            </div>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155430/Topping_Aceitunas-Negras.jpg.png?v=636766049376970000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Black Olives
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Black Olives","checked":true}
                </div>
              </div>
            </div>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155432/Topping_Jamon.jpg.png?v=636766052579200000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Ham
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Ham","checked":true}
                </div>
              </div>
            </div>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155428/Topping_Pepperoni.jpg.png?v=636766047561270000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Pepperoni
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Pepperoni","checked":true}
                </div>
              </div>
            </div>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155436/Topping_Champinones.png?v=636766057405600000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Champignon
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Champignon","checked":true}
                </div>
              </div>
            </div>
          </div>
          <div
            class="toggleChoiceContainer selected bg-washed-blue db pa4 pointer bb b--light-gray"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerToggle flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155427/Topping_Cebolla.jpg.png?v=636766046799500000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameToggle t-heading-5 c-on-base"
                  >
                    Onion
                  </div>
                </div>
              </div>
              <div
                class="toggleChoiceActions "
              >
                <div>
                  Checkbox props={"disabled":false,"name":"Onion","checked":true}
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="dn"
        >
          <div
            class="ph5 pv4 c-muted-2 t-small"
          >
            <span>
              Extra ingredients
            </span>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155432/Topping_Jamon.jpg.png?v=636766052579200000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Ham
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155428/Topping_Pepperoni.jpg.png?v=636766047561270000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Pepperoni
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155436/Topping_Champinones.png?v=636766057405600000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Champignon
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155427/Topping_Cebolla.jpg.png?v=636766046799500000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Onion
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155430/Topping_Aceitunas-Negras.jpg.png?v=636766049376970000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Black Olives
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155431/PIMIENTO-VERDE.png?v=636766050771230000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Green Pepper
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="multipleChoiceContainer false hover-bg-muted-5 w-100 ph4 pv5 bb b--muted-5"
          >
            <div
              class="relative flex justify-between items-center"
            >
              <div
                class="itemDescriptionContainer itemDescriptionContainerMultiple flex"
              >
                <div>
                  <img
                    class="br3 mr4"
                    src="https://delivery.vteximg.com.br/arquivos/ids/155435/Topping_Pina.jpg.png?v=636766056177370000"
                    style="height: 54px; width: 54px;"
                  />
                </div>
                <div
                  class="flex flex-column justify-center"
                >
                  <div
                    class="itemDescriptionName itemDescriptionNameMultiple t-heading-5 c-on-base"
                  >
                    Pineapple
                  </div>
                  <div
                    class="itemDescription itemDescriptionMultiple flex items-center c-action-primary t-small fw5 mt1"
                  >
                    <div>
                      + 
                    </div>
                    <div>
                      ProductPrice
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex-auto flex-none-ns flex justify-end"
              >
                <div
                  class="multipleChoiceActions flex-none ml4 c-action-primary t-body fw5 z-1"
                >
                  <div>
                    NumericStepper
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-100"
      />
      <div
        class="z-2 fixed bg-white bottom-0 w-40-ns ph1-ns w-100 bt b--light-gray"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductCustomizer components  render with variations example, show sku selector 1`] = `
<DocumentFragment>
  <div
    class="flex-ns pv6-ns justify-center-ns"
  >
    <div
      class="w-100 w-third-ns flex tc items-center-ns h-100-ns ph2-ns"
    >
      <img
        alt="PARRILLADA"
        class="image"
        src="https://phmex.vteximg.com.br/arquivos/ids/155565/PARRILLADA.jpg?v=636834183659030000"
        style="object-fit: cover;"
      />
    </div>
    <div
      class="w-40-ns ph2-ns"
    >
      <div
        class="productName t-heading-4 c-on-base ph5 pt5-s"
      >
        Parrillada
      </div>
      <div
        class="ph5 pt4 pb2 pb4-ns"
      />
      <div
        class="ph5"
      >
        <div>
          ExtensionPoint id="sku-selector"
        </div>
      </div>
      <div />
      <div
        class="w-100"
      />
      <div
        class="z-2 fixed bg-white bottom-0 w-40-ns ph1-ns w-100 bt b--light-gray"
      />
    </div>
  </div>
</DocumentFragment>
`;
